src/web/index.js

```diff
@@ -115,7 +124,17 @@ const params = queryString.parse(window.location.search);
 if (!params.theme) {
   postMessage('fetchTheme');
 } else {
-  jsonCodec.decompress(params.theme).then(theme => {
-    store.dispatch(actions.theme.setTheme({ theme }));
+  log('Received shared theme');
+  // TODO: figure out if this param theme matches theme from add-on and ignore if so
```

Comment:
Oh also TBD: If the ?theme param matches what the current theme already is, we should skip displaying the permission dialog. Haven't implemented that check yet.